sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -612,6 +616,12 @@ case class DeltaWithMetadataWritingSparkTask(
 private[v2] trait V2CreateTableAsSelectBaseExec extends LeafV2CommandExec {
   override def output: Seq[Attribute] = Nil
 
+  protected def commitMetrics(tableCatalog: StagingTableCatalog): Map[String, SQLMetric] = {
```

Comment:
maybe move this to `DataSourceV2Utils` as well so that we can reuse it in `ReplaceTableExec`?